src/EnergyPlus/ExtendedHeatIndex.cc

```diff
@@ -0,0 +1,552 @@
+        Real64 const flux2 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs - phi * (Tc - Tf) / Rs;
+        Real64 const flux3 = Q - Qv(Ta, Pa) - (Tc - Ta) / Ra_un(Tc, Ta) - (phi_salt * pvstar(Tc) - Pa) / Za_un;
+        if (flux1 > 0.0 && flux2 > 0.0 && flux3 >= 0.0) {
+            dTcdt = (1.0 / C) * flux3;
+        }
+        return dTcdt;
+    }
+
+    //    given T and RH, returns a key and value pair
+    Real64 find_eqvar_name_and_value(EnergyPlusData &state, Real64 const Ta, Real64 const RH, int &varname)
+    {
+        constexpr Real64 M = 83.6;                        // kg        , mass of average US adults, fryar2018
+        constexpr Real64 H = 1.69;                        // m         , height of average US adults, fryar2018
+        Real64 A = 0.202 * pow(M, 0.425) * pow(H, 0.725); // m^2       , DuBois formula, parson2014
+        constexpr Real64 cpc = 3492.;                     // J/kg/K    , specific heat capacity of core, gagge1972
+        Real64 C = M * cpc / A;                           //           , heat capacity of core
+        constexpr Real64 Q = 180.;                        // W/m^2     , metabolic rate per skin area, steadman1979
+        constexpr Real64 phi_salt = 0.9;                  //           , vapor saturation pressure level of saline solution, steadman1979
+        constexpr Real64 Tc = 310.;                       // K         , core temperature, steadman1979
+        Real64 Pc = phi_salt * pvstar(Tc);                //           , core vapor pressure
+        constexpr Real64 r = 124.;                        // Pa/K      , Zf/Rf, steadman1979
+
+        Real64 Pa = RH * pvstar(Ta);
+        Real64 Rs = 0.0387;
+        Real64 phi = 0.84;
+        Real64 dTcdt = 0.0;
+        Real64 ZsRs = Zs(Rs);
+        Real64 const m = (Pc - Pa) / (ZsRs + Za);
+        Real64 const m_bar = (Pc - Pa) / (ZsRs + Za_bar);
+
+        int SolFla;
+        Real64 Ts;
+        General::SolveRoot(
+            state,
+            tol,
+            maxIter,
+            SolFla,
+            Ts,
+            [&](Real64 Ts) { return (Ts - Ta) / Ra(Ts, Ta) + (Pc - Pa) / (ZsRs + Za) - (Tc - Ts) / Rs; },
+            std::max(0.0, std::min(Tc, Ta) - Rs * std::abs(m)),
+            std::max(Tc, Ta) + Rs * std::abs(m));
+
+        Real64 Tf;
+        General::SolveRoot(
+            state,
+            tol,
+            maxIter,
+            SolFla,
+            Tf,
+            [&](Real64 Tf) { return (Tf - Ta) / Ra_bar(Tf, Ta) + (Pc - Pa) / (ZsRs + Za_bar) - (Tc - Tf) / Rs; },
+            std::max(0.0, std::min(Tc, Ta) - Rs * std::abs(m_bar)),
+            std::max(Tc, Ta) + Rs * std::abs(m_bar));
+        Real64 const flux1 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs;
+        Real64 const flux2 = Q - Qv(Ta, Pa) - (1.0 - phi) * (Tc - Ts) / Rs - phi * (Tc - Tf) / Rs;
+        Real64 Rf;
+
+        if (flux1 <= 0.0) {
+            varname = static_cast<int>(EqvarName::Phi);
```

Comment:
Why cast this to int all the time? Just use the enum, no?

Reply:
Yeah I should use the enum class instead of casting to int